pkg/virt-controller/watch/migration.go

```diff
@@ -704,8 +706,27 @@ func (c *MigrationController) handleTargetPodCreation(key string, migration *vir
 	// migration was accepted into the system, now see if we
 	// should create the target pod
 	if vmi.IsRunning() {
-		if err := c.createPDB(key, vmi, migration); err != nil {
-			return err
+		if migrations.MigrationNeedsProtection(vmi) {
+			pdbs, err := pdbs.PDBsForVMI(vmi, c.pdbInformer)
+			if err != nil {
+				return err
+			}
+			if len(pdbs) < 1 {
+				log.Log.Object(vmi).Errorf("Found no PDB protecting the vmi")
+				return fmt.Errorf("Found no PDB protecting the vmi %s", vmi.Name)
+			}
+			pdb := pdbs[0]
```

Comment:
There is probably still a corner case.
Let's consider `PDBsForVMI` returns  2 pdbs(one from migration controller, second from disruptionbudget).
Now we pick the migration one.  `pdb := pdbs[0]` (50:50 chance).
The `expandPDB` will do nothing and `expandPDB` can be true.

In this situation, the pdb from disruptionbudget controller will not be expanded. Therefore it can happen we will not protect a migration by pdb.

Reply:
Good catch! yes actually this can happen you're right, and even worse if it happens the pdb controller will afterwards delete the pdb from the old migration controller leaving the migration unprotected.

I've pushed a new version of the PR to address this, now the migration controllers verifies that the PDB has not been generated by an old migration controller, if it has it returns an error so that the migration won't be kicked off (and the key will be re-enqueued), eventually, the pdb controller will notice this old pdb and delete it and the migration will start fine after the expansion of the correct pdb.